src/wasm/wasm-stack.cpp

```diff
@@ -2500,7 +2500,10 @@ void BinaryInstWriter::emitMemoryAccess(size_t alignment,
   if (memoryIdx > 0) {
     o << U32LEB(memoryIdx);
   }
-  o << U32LEB(offset);
+  if (memory64)
+    o << U64LEB(offset);
+  else
+    o << U32LEB(offset);
```

Comment:
Thanks for the PR!

To make sure i understand, before this PR things seemed to work because we've only ever had small-enough offsets to fit in a U32LEB, but in theory larger ones can exist and we need U64LEB encoding for that? That would explain why we've seen no errors so far. cc @sbc100 

I haven't read the spec but I'd guess this should use 32/64 encoding depending on whether the specific memory is 32/64. In LLVM there's just one global state since there's one memory, but in Binaryen we support multi-memory. If so, checking `wasm.getMemory(memory)->is64` could be what we want here.